### PR TITLE
Mr Hankey $SHIT verification

### PR DESCRIPTION
**Pull request by SoA432**
# ASCAN Token Submission

## Introduction
Before submitting this PR, ensure you have followed all the requirements below and filled out all necessary details about your token.

## Checklist
Please review the checklist below and check each item before submitting your token information:

- [x] I have reviewed the [submission guidelines](https://www.psplist.xyz/why).
- [x] The submitted files are in the correct format and fulfill all requirements outlined below.

### General Information
- **Token Name**: Mr Hankey
- **Symbol**: $SHIT
- **Contract Address**: 5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK
- **Decimals**: 12

### Logo
- [x] Logo file added to the PR
- **File Name**: `logo.png`
- **Size**: 256x256 pixels
- **File Extension**: png

### Metadata File
- [x] Metadata file added to the PR
- **File Name**: `metadata.json`
- **File Extension**: json

### Token Information File
- [x] Token information file added to the PR
- **File Name**: `info.json`
- **File Extension**: json

#### Required Content in `info.json`:
```json
{
  "name": "Mr Hankey",
  "symbol": "SHIT",
  "type": "PSP22",
  "contractAddress": "5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK",
  "decimals": 6,
  "description": "Most famous meme $SHIT",
  "website": "https://the-shit.xyz/",
  "sourceCode": "",
  "links": [
    {
      "name": "X/Twitter",
      "url": "https://x.com/mr_hankey_shit"
    },
    {
      "name": "Telegram",
      "url": "https://t.me/mr_hankey_shit"
    },
    {
      "name": "Discord",
      "url": ""
    }
  ],
  "customTags": [
    "$SHIT Contract Address"
  ],
  "warnings": [],
  "owner": "5DMCcKK5HuEtSQ9QT9eunuzLy7owhLGX5g77Es9fsoRyhcXj",
  "minter": "5DMCcKK5HuEtSQ9QT9eunuzLy7owhLGX5g77Es9fsoRyhcXj"
}

```

#### Note on Custom Tags
The `customTags` field in the info.json file is optional. This field can be used to include specific tags that describe unique characteristics or features of the token, such as burner wallet. Include custom tags only if they are relevant to the token’s functionality or usage.

#### Note on Owner and Minter
The owner and minter fields should include the addresses of the accounts that hold these capabilities within the token's smart contract. The owner is typically the account that has administrative rights over the entire contract, while the minter might be an account with the capability to issue new tokens.

### Note on Warnings
Please do not fill out the warnings field in the info.json file. This field is reserved for use by our auditing team, who will add relevant warnings such as "Mintable" or "Burnable" after thoroughly reviewing the token's smart contract. This ensures that all warnings are accurate and based on a comprehensive technical assessment.

### CODE REVIEW

**Review by DamianStraszak — CHANGES_REQUESTED** [review]
Please provide source code link.

The metadata.json does not match the on-chain contract.

**Comment by SoA432** [comment]
> Please provide source code link.
> 
> The metadata.json does not match the on-chain contract.

Hi @DamianStraszak 
Updated `sourceCode`, for metadata - can you please provide more information where we can get it ? Its just copy-paste metadata from source examples.

**Comment by DamianStraszak** [comment]
You obtain metadata after compiling the contract from source. If you didn't do it yourself, you can ask the author of the source code. If you use the INW2 contract, then the same metadata should work as in some other requests.

**Comment by SoA432** [comment]
> You obtain metadata after compiling the contract from source. If you didn't do it yourself, you can ask the author of the source code. If you use the INW2 contract, then the same metadata should work as in some other requests.

Yeap its from INW2 contract

> You obtain metadata after compiling the contract from source. If you didn't do it yourself, you can ask the author of the source code. If you use the INW2 contract, then the same metadata should work as in some other requests.

Updated

File: tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    {
+      "name": "Discord",
+      "url": ""
+    }
+  ],
+  "customTags": [
+    "$SHIT Contract Address"
+  ],
+  "warnings": [],
+  "owner": "5DMCcKK5HuEtSQ9QT9eunuzLy7owhLGX5g77Es9fsoRyhcXj",
```
Comment (DamianStraszak) [inline tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: According to my checks, the owner is a different account.

**Review by DamianStraszak — CHANGES_REQUESTED** [review]

File: tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json (hunk 1 above)
Comment (SoA432) [reply tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: Can you check once again please? I am sure address is correct according to INW2 data 

```json
{
   "_id":"6654e88f113b0a15f4ebbf50",
   "name":"Mr Hankey",
   "symbol":"SHIT",
   "decimal":12,
   "contractAddress":"5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK",
   "tokenGeneratorContractAddress":"5CWKnYhJr7jFw7a2ojKhdL3iBBGYohkoSzQQnj5v7AjmDiWu",
   "creator":"5DMCcKK5HuEtSQ9QT9eunuzLy7owhLGX5g77Es9fsoRyhcXj",
   "mintTo":null,
   "tokenIconUrl":"QmVVwDAy9VVRfbhWqhRD4rrvUV2JDCEy5ShC9Ko7U1Vgj3",
   "totalSupply":21000000,
   "index":20,
   "isManagedByTokenGenerator":true,
   "createdTime":"2024-05-27T20:09:51.234Z",
   "updatedTime":"2024-07-16T21:42:31.279Z"
}
```

Comment (SoA432) [reply tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: Or owner should be INW2 contract address ?

Comment (DamianStraszak) [reply tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: Owner is the owner -- the entity who can mint and do whatever they want to the contract. That's what the contract code indicates. If you didn't set-up the contract by yourself please ask for support whoever did it for you.

Comment (SoA432) [reply tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: @DamianStraszak I cant understand where did you get this information ? What owner's address you see? I am 100% sure its same as minter `5DMCcKK5HuEtSQ9QT9eunuzLy7owhLGX5g77Es9fsoRyhcXj`

Comment (DamianStraszak) [reply tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: With all due respect, but you should not be asking me who the owner of your contract is, or how to get that information.

This form is to collect facts and notarize them, not to get you technical support.

Comment (SoA432) [reply tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: Okay thanks for support, will no more wasting time on Azero

Comment (SoA432) [reply tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: @DamianStraszak prove me i am wrong
Снимок экрана 2024-07-17 в 11 10 44

Comment (DamianStraszak) [reply tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: This is being discussed with the InkWhale team now. Will let you know soon.

Comment (DamianStraszak) [reply tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: Alright. The issue is that you provided incorrect metadata.json file. The owner is specified correctly, but I was getting incorrect results because the provided metadata did not match the contract. 

I think that the correct metadata file for your contract is https://github.com/InkWhale-net/contracts/blob/feature/upgrade-op-4/op4_contracts/contracts/psp22_standard/psp22_standard.json -- but I don't know that, it's only speculation. So I would encourage you to ask InkWhale to confirm that.

So, long story short -- the metadata.json needs to be corrected in the submission.

Comment (SoA432) [reply tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: @DamianStraszak got new information that may can help: When token was created and minted there was AZERO.ID under owner account, after creation/mint azero.id was removed, maybe this create somekind of missmatch under owner account ?

Comment (SoA432) [reply tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: @DamianStraszak yeah as we discussed previously metadata was just copy-pasted from other PR's and i took 'debug' one and didnt notice it. Updated now with 'release' one

Comment (DamianStraszak) [reply tokens/psp22/5F9aiiwLMPC6fFxxwqHvJpm7h5T4Xm93mJT6cpDrQnKkLFoK/info.json:26]: Alright. I checked the new version a everything is fine.

**Review by DamianStraszak — APPROVED** [review]
